packages/pipeline/test/data_sources/ohlcv_external/crypto_compare_test.ts

```diff
@@ -0,0 +1,51 @@
+import * as chai from 'chai';
+import 'mocha';
+import { merge } from 'ramda';
+
+import { CryptoCompareOHLCVSource } from '../../../src/data_sources/ohlcv_external/crypto_compare';
+import { chaiSetup } from '../../utils/chai_setup';
+
+chaiSetup.configure();
+const expect = chai.expect;
+
+// tslint:disable:custom-no-magic-numbers
+describe('ohlcv_external data source (Crypto Compare)', () => {
+    describe('generateBackfillIntervals', () => {
+        it('generates pairs with intervals to query', () => {
+            const source = new CryptoCompareOHLCVSource();
+            const pair = {
+                fromSymbol: 'ETH',
+                toSymbol: 'ZRX',
+                latest: new Date().getTime() - source.interval * 2,
+            };
+
+            const expected = [
+                pair,
+                merge(pair, { latest: pair.latest + source.interval }),
+                merge(pair, { latest: pair.latest + source.interval * 2 }),
+            ];
+
+            const actual = source.generateBackfillIntervals(pair);
+            expect(actual).deep.equal(expected);
+        });
+
+        it('returns single pair if no backfill is needed', () => {
+            const source = new CryptoCompareOHLCVSource();
+            const pair = {
+                fromSymbol: 'ETH',
+                toSymbol: 'ZRX',
+                latest: new Date().getTime() - source.interval + 5000,
+            };
+
+            const expected = [pair];
+
+            const actual = source.generateBackfillIntervals(pair);
+            expect(actual).deep.equal(expected);
+        });
+    });
+
+    describe('getAsync', () => {
+        // tslint:disable:no-empty
+        // todo
```

Comment:
Still open TODO?

Reply:
We don't have any infra/tooling in place to test functions which send HTTP requests. I think the plan is to eventually use something like [nock](https://www.npmjs.com/package/nock).

We should either clarify this TODO comment or just remove the `getAsync` test. When we add `nock` or some other tooling we can go through and add similar tests for the other files in __data_sources/__.